Content.Shared/HeadSlime/SharedHeadSlimeSystem.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Content.Shared.Movement.Systems;
+using Content.Shared.DoAfter;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Mobs;
+using Robust.Shared.Containers;
+using Content.Shared.Actions;
+using Content.Shared.Popups;
+using Robust.Shared.Serialization;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Humanoid;
+
+namespace Content.Shared.HeadSlime;
+
+public abstract class SharedHeadSlimeSystem : EntitySystem
+{
+    /// <inheritdoc/>
+    [Dependency] private readonly SharedDoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<HeadSlimeComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshSpeed);
+
+        SubscribeLocalEvent<HeadSlimeComponent, HeadSlimeInfectEvent>(OnHeadSlimeInfectAction);
+        SubscribeLocalEvent<HeadSlimeComponent, HeadSlimeInjectEvent>(OnHeadSlimeInjectAction);
+    }
+
+    private void OnRefreshSpeed(EntityUid uid, HeadSlimeComponent component, RefreshMovementSpeedModifiersEvent args)
+    {
+        if(!component.HeadSlimeQueen)
+        {
+            var mod = component.HeadSlimeMovementSpeedDebuff;
+            args.ModifySpeed(mod, mod);
+        }
+    }
+
+    private void OnHeadSlimeInfectAction(EntityUid uid, HeadSlimeComponent component, HeadSlimeInfectEvent args)
+    {       
+        if (args.Handled)
+            return;
+
+        args.Handled = true;
+        var target = args.Target;
+
+        if (HasComp<HeadSlimeComponent>(target))
+            return;
+
+        if (!HasComp<HumanoidAppearanceComponent>(target))
+        {
+            _popupSystem.PopupEntity(Loc.GetString("Head-Slime-action-popup-message-fail-target-animal"), uid, uid);
+            return;
+        }
```

Comment:
This will trigger (thus telling the player that "this cannot be used on an animal") even if the action is used on an inanimate object like a toolbox.